### PR TITLE
chore: bunify & tidy up internal/user

### PR DESCRIPTION
**Pull request by carolinaecalderon**
## Description
Related to  https://github.com/determined-ai/determined/pull/7769, this bunifies & tidies up `master/internal/user/postgres_agentusergroup.go` and moves the methods in `master/internal/user/external_users.go` to `postgres_users.go`

## Test Plan
Create a user & experiment & test getting its agent user group.

Add tests postgres_agentusergroup.go as part of https://github.com/determined-ai/determined/pull/7875

For EE: circle CI checks (https://app.circleci.com/pipelines/github/determined-ai/determined-ee?branch=postgres-agent-user-groups), test branch (https://github.com/determined-ai/determined-ee/tree/postgres-agent-user-groups)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
DET-9841

### CODE REVIEW

File: master/internal/user/external_users.go [hunk 1]
```diff
@@ -1,12 +0,0 @@
-package user
-
-import (
-	"github.com/determined-ai/determined/master/pkg/model"
-)
-
-// UserByExternalToken returns a user session derived from an external authentication token.
-func UserByExternalToken(tokenText string,
```
Comment (stoksc) [inline master/internal/user/external_users.go:8]: this is filled in in ee, in case you didn't see.

**Review by stoksc — APPROVED** [review]
lgtm just make sure you are ready to fix ee

File: master/internal/user/external_users.go (hunk 1 above)
Comment (carolinaecalderon) [reply master/internal/user/external_users.go:8]: will be included in my EE fix

File: master/internal/user/postgres_agentusergroup.go [hunk 2]
```diff
@@ -37,31 +20,51 @@ type optionalAgentUserGroup struct {
...
 	}
 
-	err := db.Bun().NewRaw(`
-SELECT
-	uid, user_ as user, gid, group_ as group
-FROM workspaces JOIN projects ON workspaces.id = projects.workspace_id
-WHERE projects.id = ?`,
-		e.ProjectID).Scan(context.TODO(), &aug)
+	err := db.Bun().NewSelect().Table("workspaces").
+		ColumnExpr("uid, user_ AS user, gid, group_ AS group").
```
[the remark this answers is not in the document]
Comment (carolinaecalderon) [reply master/internal/user/postgres_agentusergroup.go:50]: okay it's becoming more apparent/annoying that the agentUserGroup ID should _at least_ be stored in the experiments struct/table, if not the projects table -- what do you think about this?

[the remark this answers is not in the document]
Comment (stoksc) [reply master/internal/user/postgres_agentusergroup.go:50]: not sure exactly what you mean. want to slack huddle today?

File: master/internal/user/postgres_users.go [hunk 3]
```diff
@@ -327,6 +327,13 @@ func ByToken(ctx context.Context, token string, ext *model.ExternalSessions) (
 	return &user, &session, nil
 }
 
+// ByExternalToken returns a user session derived from an external authentication token.
```
[the remark this answers is not in the document]
Comment (carolinaecalderon) [reply master/internal/user/postgres_users.go:330]: running nils on OSS, but in EE it's filled out

[the remark this answers is not in the document]
Comment (stoksc) [reply master/internal/user/postgres_users.go:330]: side note: put this in its own file to minimize rebase conflicts.

**Pull request merged**